### DISCUSSION TITLE
BIP39 Restore broken

### DISCUSSION THREAD

**Discussion by 3point14159** [horizontalsystems/unstoppable-wallet-android#5560] [General]
I [already filed an issue for the broken BIP39 key generation](https://github.com/horizontalsystems/unstoppable-wallet-android/issues/5558) in the wallet, but wanted to start a thread here also.

My scenario is as such: non-english seed phrase + BIP39 passphrase (with heavy use of special characters).

You can test the output from your broken code against the key generation here:

https://github.com/iancoleman/bip39

**Comment by esengulov** [comment]
noted, debugging it now. We were indeed able to reproduce the issue for some cases where words included diacritic characters.

**Comment by 3point14159** [comment]
@esengulov and "Unstoppable" team,

The big question here is what is the point of issues sitting in "testing" if they aren't actually being tested? I was forgiving of the long release cycles because I assumed (I also think it was later explained as such as well) that these issues needed time to be tested after coding changes. Another example of poor quality assurance. I even tried to volunteer to help with such issues, no one will help you help yourselves and bring someone who cares aboard.

You should really consider renaming this app, the irony of the name stands in embarrassing contrast to the reality of the functionality of the app. It has been impossible to use your app now for three full months for the same issue. Just ditch the testing/review section on the project tracker, we all know it isn't being used, and start making shorter release cycles.

**Comment by 3point14159** [comment]
Thanks for the recent release, keep em coming!

https://github.com/horizontalsystems/unstoppable-wallet-android/releases/tag/0.28.1

After reading the notes, I am confident the devs here can figure out how to do a carbon-copy (1:1) port of special characters between foreign derived keys. This is something that should have been tested before release, assuming there is anything of a testing phase. I would love to test drive beta versions to help with much needed quality assurance before "stable" releases are made on the main release channel.

**Reply by esengulov** [reply]
hey there,

It would be great if you participate as a tester. Your feedback so far prooved invaluable) We are working on some of the mentioned points and are evolving as a result. We will add you as a beta tester to the App Center so you can follow our features as we progress and talk through some of the stuff as it happens.

Our android lead is @abdrasulov. Generally, major implementation decisions are coordinated between @ealymbaev @abdrasulov. That is in case you wonder whom to address specifically should there be a need for it.

it's much appreciated that you're sticking around!

**Reply by Dianaaiym** [reply]
> Thanks for the recent release, keep em coming!
> 
> https://github.com/horizontalsystems/unstoppable-wallet-android/releases/tag/0.28.1
> 
> After reading the notes, I am confident the devs here can figure out how to do a carbon-copy (1:1) port of special characters between foreign derived keys. This is something that should have been tested before release, assuming there is anything of a testing phase. I would love to test drive beta versions to help with much needed quality assurance before "stable" releases are made on the main release channel.

Hi,

Please write to us at hsdao@protonmail.com, and we'll add you as a beta tester in the App Center.

**Reply by 3point14159** [reply]
Awesome. Thanks for looping me in, I will do my best to provide meaningful feedback from the fringes of functionality. I would love to see this wallet succeed where some many other wallets have fallen short, but still lots more work to-do.

**Comment by 3point14159** [comment]
Great news to report; v.28.2 worked for me. Non-english bip39's with special characters seems to be solved at the moment. Will test it further, but in the meantime, please push an upgrade package to f-droid as soon as you can!

Also, I hope this feature can remain in the core app, I remember reading (something to the effect of) that if this didn't work out that it may be dropped as a feature all together. That would be unfortunate considering the work that was put into getting it working finally. Keep it in please.

**Comment by abdrasulov** [comment]
Great to hear that. We're glad to inform you that version 0.28.2 is available on f-droid too.

**Comment by 3point14159** [comment]
Excellent progress. One small step for privacy, and a giant leap towards a freer world. Cheers!
